### ISSUE TITLE
[Discussion] Adding sklearn dependency for data normalization

### ISSUE THREAD

**Issue by sxjscience** [dmlc/gluon-nlp#1290]
According to the offline discussion with Jonas (@jwmueller), we are considering to reuse features from sklearn, like https://scikit-learn.org/stable/modules/generated/sklearn.impute.SimpleImputer.html#sklearn.impute.SimpleImputer.transform for basic data normalization in the upcoming AutoML feature.

The problem here is whether we should add the dependency of sklearn to GluonNLP.

According to the web-page, Scikit-learn has been used in many companies so it should not be a very big issue: https://scikit-learn.org/stable/testimonials/testimonials.html.

Would like to listen to your opinion: @dmlc/gluon-nlp-committers

**Comment by szha** [comment]
In GluonNLP, we have the goal of making the data pipeline portable to other platforms, e.g. outside of python. Would this design choice affect that goal?

**Comment by sxjscience** [comment]
Basically, we may write the inference functions manually (e.g., via some C++ codes). My concern is that I'm not sure if sklearn is memory-efficient or can deal with large-scale dataset.
